### PR TITLE
sst_importer: add import mode timeout

### LINKED ISSUE #7223: UCP: Automatically dismiss "Import mode"
## Description

When using TiDB Lightning or BR to ingest data into a TiKV cluster, these tools will place every TiKV node into an "[import mode](https://github.com/tikv/tikv/blob/master/components/sst_importer/src/import_mode.rs)" to speed up writing (while sacrificing storage space). 

Normally, when these tools have completed operation, they will instruct TiKV to move back to "normal mode". However, when Lightning or BR crashed, the TiKV nodes will be stuck in "import mode" forever until we manually switch back (which not many people will remember to do so).

To implement this enhancement, you should

* Set up a timer. When time's up, change the TiKV back into normal mode. Reset the timer when the `sst_importer` client actively request staying in import mode.
    * Lightning already sends a "switch to import mode" continuously every 5 minutes. You don't need to care about fixing Lightning/BR in this issue.
* Add a configuration for the timer's duration.
* Include a test case showing that a TiKV does indeed move back to "normal mode" after a certain time period.

## Score

* 605

## Mentor(s)

* @kennytm

Contact the mentors: **#tidb-challenge-program** channel in [TiDB Community](https://join.slack.com/t/tidbcommunity/shared_invite/enQtNzc0MzI4ODExMDc4LWYwYmIzMjZkYzJiNDUxMmZlN2FiMGJkZjAyMzQ5NGU0NGY0NzI3NTYwMjAyNGQ1N2I2ZjAxNzc1OGUwYWM0NzE)  Slack Workspace

### PR DESCRIPTION
**Pull request by codeworm96**
UCP #7223 

### What problem does this PR solve?

Issue Number: close #7223

Problem Summary: Currently, if Lightning or BR crashes, the TiKV nodes will be left in the import mode until the user manually switch them back into the normal mode.

### What is changed and how it works?

What's Changed: In `ImportModeSwitcher`, a background future is added to put TiKV back into normal mode after the timeout.

### Check List 

Tests 

- Unit test

Side effects
- A new configuration for import mode timeout is added.

### Release note

### CODE REVIEW

**Comment by overvenus** [comment]
PTAL @kennytm

File: src/import/sst_service.rs [hunk 1]
```diff
@@ -61,13 +61,14 @@ impl<Router: RaftStoreRouter<RocksSnapshot>> ImportSSTService<Router> {
             .name_prefix("sst-importer")
             .pool_size(cfg.num_threads)
             .create();
+        let switcher = ImportModeSwitcher::new(&cfg, &threads, engine.clone());
```
Comment (overvenus) [inline src/import/sst_service.rs:64]: Could you create it on demand?

File: components/sst_importer/src/import_mode.rs [hunk 2]
```diff
@@ -1,67 +1,143 @@
...
 use kvproto::import_sstpb::*;
+use tikv_util::timer::GLOBAL_TIMER_HANDLE;
 
+use super::Config;
 use super::Result;
 
 type RocksDBMetricsFn = fn(cf: &str, name: &str, v: f64);
 
-pub struct ImportModeSwitcher {
+struct ImportModeSwitcherInner<T: KvEngine> {
```
Comment (kennytm) [inline components/sst_importer/src/import_mode.rs:19]:
```suggestion
struct ImportModeSwitcherInner<E: KvEngine> {
```

TiKV typically uses `E` for the engine.

File: components/sst_importer/src/import_mode.rs [hunk 3]
```diff
@@ -220,15 +298,55 @@ mod tests {
...
+            .name_prefix("sst-importer")
+            .pool_size(cfg.num_threads)
+            .create();
+
+        let mut switcher = ImportModeSwitcher::new(&cfg, &threads, db.clone());
+        check_import_options(&db, &normal_db_options, &normal_cf_options);
+        switcher.enter_import_mode(mf).unwrap();
+        check_import_options(&db, &import_db_options, &import_cf_options);
+
+        thread::sleep(Duration::from_secs(10));
```
Comment (kennytm) [inline components/sst_importer/src/import_mode.rs:348]: Please use a shorter duration for this test (e.g. 300ms timeout and sleep for 1s)

**Pull request merged**